src/aggregator/aggregator/aggregator.go

```diff
@@ -198,13 +198,13 @@ func (agg *aggregator) placementTick() {
 			return
 		}
 
-		agg.RLock()
 		placement, err := agg.placementManager.Placement()
 		if err != nil {
 			m.updateFailures.Inc(1)
 			continue
 		}
 
+		agg.RLock()
```

Comment:
@vdarulis  I *think* the move of the lock here is safe, because nothing above touches state on `agg` itself (and `placementManager` has its own lock), but definitely check me on that.

Reply:
I think so, unless tests do something crazy and fail w/ race cond.

Reply:
Cool yeah that's my read as well, thanks!